### ISSUE TITLE
Understanding of derivatives with matrices/vectors in back propagation. 

### ISSUE THREAD

**Issue by ZelongZeng** [ZelongZeng/Blog#2]
Recently, I'm studying something about RNN because I want to access the field of action recognition in video. So I found that many people don't understand the meaning of derivatives with matrices, because many tutorials use this form to present the process of BP(Back Propagation). In fact, we don't need to understand the mathematical meaning of it, I mean, most of people use it because it can let the expression show shorter and easier. 
  For example. In RNN, we can to compute:
![latex-1](https://user-images.githubusercontent.com/40358919/44158431-9ad3cd80-a0ef-11e8-9ee8-c806c6c4668a.png)
  E3 is a scalar and V is a matrix. E3 is the cross entropy loss and V is a parameter, so we want to update the V to get a better result. This example is scalar-by-matrix. But in fact, we don't need to care the original meaning of this kind of derivative. We know, in the process of BP, people use Gradient Descent to optimize parameters. The direction of gradient is: 
<img width="207" alt="wx20180816-005439 2x" src="https://user-images.githubusercontent.com/40358919/44158105-fe113000-a0ee-11e8-94a6-f717512b5702.png">
  So we just regard each element of V as variate of E3, and compute the gradient like the above equation. That's easy to understand, right?
  For the matrix-by-matrix, let:
<img width="80" alt="wx20180816-011201 2x" src="https://user-images.githubusercontent.com/40358919/44159108-66f9a780-a0f1-11e8-8c39-e214a96d694f.png">
 is
<img width="276" alt="wx20180816-010935 2x" src="https://user-images.githubusercontent.com/40358919/44158953-108c6900-a0f1-11e8-8d4a-e00a3178ca69.png">
  we can get:
<img width="164" alt="wx20180816-011026 2x" src="https://user-images.githubusercontent.com/40358919/44158993-2c900a80-a0f1-11e8-847b-fc4374dcb604.png">
  So:
<img width="200" alt="wx20180816-011102 2x" src="https://user-images.githubusercontent.com/40358919/44159032-43366180-a0f1-11e8-952f-70aa6419167c.png">
  The form of result is not important, it is up to you. 

  Finally, there is a good tutorial about RNN: http://www.wildml.com/category/rnns/
  Recommend it to all.
